### ISSUE TITLE
help with custom attributes

### ISSUE THREAD

**Issue by bhenderson** [go-cas/cas#2]
hi there,

my work's CAS server is returning a response like

```xml
<?xml version="1.0" encoding="UTF-8"?>                                                                
<cas:serviceResponse xmlns:cas="http://www.yale.edu/tp/cas">                                          
  <cas:authenticationSuccess>
    <cas:user>bhenderson</cas:user>                                                                   
    <cas:attributes>                                                                                  
      <cas:user_id>1234</cas:user_id>
      <cas:login>bhenderson</cas:login>
      <cas:name>Brian Henderson</cas:name>                                                            
      <cas:email>me@example.com</cas:email>                                                           
      <cas:reports_to>myboss</cas:reports_to> 
    </cas:attributes>
    <cas:attribute name="user_id" value="1234"/>                                                      
    <cas:attribute name="login" value="bhenderson"/>                                                  
    <cas:attribute name="name" value="Brian Henderson"/>                                              
    <cas:attribute name="email" value="me@example.com"/> 
    <cas:attribute name="reports_to" value="myboss"/>                                                 
  </cas:authenticationSuccess>
</cas:serviceResponse>
```

and the extra attributes aren't getting picked up. Do you have any suggestions? I don't know a lot about what a correct response should look like, so I don't know if my company is wrong (our CAS server is old) or if the spec is just really broad.

thanks,

**Comment by geoffgarside** [comment]
I've merged your PR and released v1.0.3. Your CAS XML doesn't completely match up to the latest specifications, or ones I've looked at in the past. The `cas:attribute` elements you've got shouldn't exist directly within the `cas:authenticationSuccess` element, as far as I've seen, and should be contained within the `cas:attributes` element as your others are. It could be that your CAS server is doing both in order to work with CAS clients supporting different versions of the specification.

Are you using the official JASIG CAS server, a third party, or an in-house one? We use an in-house one at $work which can make things easier and harder at the same time.
